pkg/controllers/federate/util.go

```diff
@@ -79,21 +81,33 @@ func newFederatedObjectForSourceObject(
 	observedLabelKeys := generateObservedKeys(sourceObj.GetLabels(), federatedLabels)
 
 	federatedAnnotations, templateAnnotations := classifyAnnotations(sourceObj.GetAnnotations())
```

Comment:
This variable seems redundant. Let's inline it and LGTM.

Reply:
done